pandas/core/arrays/integer.py

```diff
@@ -206,7 +206,17 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             mask = mask.copy()
         return values, mask
 
-    values = np.array(values, copy=copy)
+    if hasattr(values, "dtype"):
+        if isinstance(values, Categorical):
+            values = np.array(values, copy=copy)
```

Comment:
What's the reason for the categorical special case? Did you run into a failing test for categoricals? (can you show an example?)

Reply:
When passing `Categorical data` to `integer_array` it generates `TypeError: category cannot be converted to an IntegerDtype` because of not converting it to numpy array. So Categoricals must be converted to numpy array.

Reply:
OK, but you can then do `values = np.array(values, copy=copy)` in the `hasattr(values, "dtype")` branch for all cases, also for the non-categorical case this will do the correct thing